qutip/core/cy/qobjevo.pyx

```diff
@@ -855,19 +794,24 @@ cdef class QobjEvo:
     @property
     def isoper(self):
         """Indicates if the system represents an operator."""
-        # TODO: isoper should be part of dims
-        if self._isoper == -1:
-            self._isoper = type_from_dims(self.dims) == "oper"
-        return self._isoper
+        return self._dims.type == "oper"
 
     @property
     def issuper(self):
         """Indicates if the system represents a superoperator."""
-        # TODO: issuper should/will be part of dims
-        # remove self._issuper then
-        if self._issuper == -1:
-            self._issuper = type_from_dims(self.dims) == "super"
-        return self._issuper
+        return self._dims.issuper
+
+    @property
+    def dims(self):
+        return self._dims.as_list()
```

Comment:
I'd also be happy to only use the list format for input and printing the dims, but it would break some existing usage patterns (e.g. `.dims[0]`). This is v5 though, so maybe that is okay. Perhaps something we can discuss in the admin meeting or on the mailing list?